packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/repository.ts

```diff
@@ -601,39 +736,64 @@ export class SavedObjectsRepository implements ISavedObjectsRepository {
     objects: SavedObjectsCheckConflictsObject[] = [],
     options: SavedObjectsBaseOptions = {}
   ): Promise<SavedObjectsCheckConflictsResponse> {
+    const namespace = this.getCurrentNamespace(options.namespace);
+
     if (objects.length === 0) {
       return { errors: [] };
     }
 
-    const namespace = normalizeNamespace(options.namespace);
-
     let bulkGetRequestIndexCounter = 0;
-    const expectedBulkGetResults: Array<Either<Record<string, any>, Record<string, any>>> =
-      objects.map((object) => {
-        const { type, id } = object;
-
-        if (!this._allowedTypes.includes(type)) {
-          return {
-            tag: 'Left',
-            value: {
-              id,
-              type,
-              error: errorContent(SavedObjectsErrorHelpers.createUnsupportedTypeError(type)),
-            },
-          };
-        }
+    type ExpectedBulkGetResult = Either<
+      { type: string; id: string; error: Payload },
+      { type: string; id: string; esRequestIndex: number }
+    >;
+    const expectedBulkGetResults = objects.map<ExpectedBulkGetResult>((object) => {
+      const { type, id } = object;
 
+      if (!this._allowedTypes.includes(type)) {
         return {
-          tag: 'Right',
+          tag: 'Left',
           value: {
-            type,
             id,
-            esRequestIndex: bulkGetRequestIndexCounter++,
+            type,
+            error: errorContent(SavedObjectsErrorHelpers.createUnsupportedTypeError(type)),
           },
         };
+      }
+
+      return {
+        tag: 'Right',
+        value: {
+          type,
+          id,
+          esRequestIndex: bulkGetRequestIndexCounter++,
+        },
+      };
+    });
+
+    const namespaceString = SavedObjectsUtils.namespaceIdToString(namespace);
+    const validObjects = expectedBulkGetResults.filter(isRight);
+    const typesAndSpaces = new Map<string, Set<string>>();
+    for (const { value } of validObjects) {
+      typesAndSpaces.set(value.type, new Set([namespaceString])); // Always enforce authZ for the active space
+    }
+    const authorizationResult = await this._securityExtension?.checkAuthorization({
+      types: new Set(typesAndSpaces.keys()),
+      spaces: new Set([namespaceString]), // Always check authZ for the active space
+      actions: ['bulk_create'],
+    });
+    if (authorizationResult) {
+      this._securityExtension!.enforceAuthorization({
+        typesAndSpaces,
+        action: 'bulk_create',
+        typeMap: authorizationResult.typeMap,
+        // auditCallback is intentionally omitted, this function in the previous Security SOC wrapper implementation
+        // did not have audit logging. This is primarily because it is only used internally during imports, it is not
+        // exposed in a public HTTP API
```

Comment:
`checkConflicts` is used in https://github.com/elastic/kibana/blob/914d1cd41c87e34df4144032603b99a281c31ba6/x-pack/plugins/actions/server/lib/connector_token_client.ts and we have no indication that this is a private API that shouldn't be used. 

Maybe we never care to audit this function but the reason given doesn't seem correct.

Reply:
I think the comment is misleading. My understanding is that this function is not meant to be private to the saved objects packages, but rather intended for use internal to Kibana only.